reference-architectures/sandboxes/sandbox-modules/tf/system-project/main.tf

```diff
@@ -0,0 +1,317 @@
+module "terraform_bucket" {
+  source = "../../../fabric-modules/gcs"
+
+  project_id = module.system_project.id
+  name       = "${module.system_project.name}_terraform"
+  location   = "US"
+  versioning = true
+}
+
+module "terraform_state_bucket" {
+  source = "../../../fabric-modules/gcs"
+
+  project_id = module.system_project.id
+  name       = "${module.system_project.name}_terraform_state"
+  location   = "US"
+  versioning = true
+}
+
+resource "null_resource" "upload_tf" {
+  provisioner "local-exec" {
+    command = <<EOT
+      find ${abspath("${path.module}/../../../catalog")} -type f -name '*.tf' -exec sed -i 's/TERRAFORM_GCS_BUCKET/${module.terraform_bucket.name}/g' {} +
+      gcloud storage rsync ${abspath("${path.module}/../../../fabric-modules")} gs://${module.terraform_bucket.name}/fabric-modules/ --recursive --delete-unmatched-destination-objects
+      gcloud storage rsync ${abspath("${path.module}/../../../catalog")} gs://${module.terraform_bucket.name}/catalog/ --recursive
+    EOT
+  }
+
+  triggers = {
+    always_run = "${timestamp()}"
+  }
+}
+
+module "sandbox_factory_sa" {
+  source = "../../../fabric-modules/iam-service-account"
+
+  project_id = module.system_project.id
+  name       = "sandbox-factory-sa"
+
+  iam_billing_roles = {
+    "${var.billing_account}" = [
+      "roles/billing.user"
+    ]
+  }
+
+  iam_folder_roles = {
+    "${var.sandboxes_folder}" = [
+      "roles/resourcemanager.projectCreator",
+      "roles/resourcemanager.projectDeleter"
+    ]
+  }
+
+  iam_project_roles = {
+    "${module.system_project.id}" = [
+      "roles/cloudbuild.builds.builder",
+      "roles/config.admin",
+      "roles/storage.admin"
+    ]
+  }
+}
+
+#
+# Firestore
+#
+
+resource "google_firestore_database" "database" {
+  project     = var.system_project_name
+  name        = "(default)"
+  location_id = "nam5"
+  type        = "FIRESTORE_NATIVE"
+}
+
+#
+# Cloud Run
+#
+
+resource "google_artifact_registry_repository" "infra_manager_processor" {
+  project       = module.system_project.id
+  location      = "us-central1"
+  repository_id = "infra-manager-processor"
+  description   = "Docker Image repository for infra-manager-processor Cloud Run service"
+  format        = "DOCKER"
+}
+
+resource "null_resource" "build_image" {
+  provisioner "local-exec" {
+    command = <<EOT
+      cd ${abspath("${path.module}/../../src/infra-manager-processor")}
+      gcloud builds submit --project=${module.system_project.id} --tag us-central1-docker.pkg.dev/${module.system_project.id}/${google_artifact_registry_repository.infra_manager_processor.repository_id}/infra-manager-processor
+    EOT
+  }
+
+  triggers = {
+    always_run = "${timestamp()}"
+  }
+
+  depends_on = [google_artifact_registry_repository.infra_manager_processor]
+}
+
+module "cloud_run" {
+  source     = "../../../fabric-modules/cloud-run-v2"
+  project_id = module.system_project.id
+  name       = "infra-manager-processor"
+  region     = "us-central1"
```

Comment:
should this come form the config?

Reply:
Yes it should, I had intended to capture that in a TODO.  TODOs updated.

Reply:
Changing the region can now be done through a variable.